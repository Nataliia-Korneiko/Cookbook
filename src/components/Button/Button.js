/* eslint-disable react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';
import s from './Button.module.css';

const Button = ({ type = 'button', label = '', onClick = () => null }) => {
  return (
    <button className={s.button} type={type} onClick={onClick}>
      {label}
    </button>
  );
};

Button.propTypes = {
  type: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Button;
