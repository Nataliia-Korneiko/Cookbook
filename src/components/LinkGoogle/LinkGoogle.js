import React from 'react';
import { ReactComponent as IconGoogle } from '../../assets/icons/icon-google.svg';
import s from './LinkGoogle.module.css';

const LinkGoogle = () => {
  return (
    <a className={s.link__google} href="https://www.google.com.ua/">
      <div className={s.link__wrapper}>
        <IconGoogle width="28" height="28" />
        <span className={s.link__span}>Google</span>
      </div>
    </a>
  );
};

export default LinkGoogle;
