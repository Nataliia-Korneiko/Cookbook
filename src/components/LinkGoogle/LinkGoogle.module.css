.link__google {
  display: block;
  width: 120px;
  height: 44px;
  border: 2px solid var(--color-golden);
  border-radius: 5px;
  font-family: 'Montserrat', sans-serif;
  font-size: 18px;
  font-weight: 500;
  color: var(--color-black);
  background-color: var(--bg-color-white);
  box-shadow: var(--box-shadow-light);
  outline: none;
  transition: 0.8s ease-in-out;
  box-shadow: var(--box-shadow-light);
}

.link__google:hover,
.link__google:focus {
  background-color: var(--bg-color-golden);
  transition: 0.8s ease-in-out;
}

.link__wrapper {
  display: flex;
  padding-left: 8px;
  padding-top: 6px;
}

.link__span {
  padding-left: 6px;
  line-height: 28px;
}
