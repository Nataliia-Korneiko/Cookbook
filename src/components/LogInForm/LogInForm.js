import React, { Component } from 'react';
// import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import shortid from 'shortid';
import routes from '../../routes/routes';
import LinkGoogle from '../LinkGoogle/LinkGoogle';
import Button from '../ButtonAuthorization/ButtonAuthorization';
import s from './LogInForm.module.css';

class LogInForm extends Component {
  state = {
    email: '',
    password: '',
    error: null,
  };

  ids = {
    emailId: shortid.generate(),
    passwordId: shortid.generate(),
  };

  render() {
    const { email, password, error } = this.state;
    return (
      <div className={s.login}>
        <LinkGoogle />
        <form
          className={s.form}
          onSubmit={this.handleSubmit}
          autoComplete="off"
        >
          <label className={s.label} htmlFor={this.ids.emailId}>
            E-mail&#58;
            <input
              className={s.input}
              id={this.ids.emailId}
              type="email"
              name="email"
              value={email}
              onChange={this.handleChange}
              placeholder="Enter your email"
            />
            {error && <span className={s.error}>{error.email}</span>}
          </label>
          <label className={s.label} htmlFor={this.ids.passwordId}>
            Password&#58;
            <input
              className={s.input}
              id={this.ids.passwordId}
              type="password"
              name="password"
              value={password}
              onChange={this.handleChange}
              placeholder="Enter your password"
            />
            {error && <span className={s.error}>{error.password}</span>}
          </label>
          <Button label="Log in" type="submit" />
          <p className={s.relink}>Don&#39;t have an account?</p>
          <NavLink to={routes.SIGN_UP} className={s.link}>
            Sign Up
          </NavLink>
        </form>
      </div>
    );
  }
}

export default LogInForm;
