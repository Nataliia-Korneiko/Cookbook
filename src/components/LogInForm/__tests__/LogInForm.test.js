import React from 'react';
import { shallow } from 'enzyme';
import { NavLink } from 'react-router-dom';
import LogInForm from '../LogInForm';
import Button from '../../ButtonAuthorization/ButtonAuthorization';
import LinkGoogle from '../../LinkGoogle/LinkGoogle';

describe('LogInForm component', () => {
  const initialState = {
    email: '',
    password: '',
    error: null,
  };

  const newLogInForm = shallow(<LogInForm />);

  it('initialize initialState with initial state', () => {
    expect(newLogInForm.state()).toEqual(initialState);
  });

  it('renders a properly', () => {
    expect(newLogInForm).toMatchSnapshot();
  });

  // -- Does not work!!! --
  // let wrapped;

  // beforeEach(() => {
  //   wrapped = shallow(<LogInForm />);
  // });

  // it('renders inputs and a button', () => {
  //   expect(wrapped.find('input[type="email"]').length).toBe(1);
  //   expect(wrapped.find('input[type="password"]').length).toBe(1);
  //   expect(wrapped.find(Button).length).toBe(1);
  // });

  // describe('events', () => {
  //   const emailText = 'new email text';

  //   beforeEach(() => {
  //     const changeEventMock = {
  //       target: {
  //         value: emailText,
  //       },
  //     };

  //     wrapped.find('input[type="email"]').simulate('change', changeEventMock);
  //     wrapped.update();
  //   });

  //   it('can enter text into an input', () => {
  //     expect(wrapped.find('input[type="email"]').prop('value')).toBe(emailText);
  //   });

  //   it('on form submit, input should get emptied', () => {
  //     expect(wrapped.find('input[type="email"]').prop('value')).toBe(emailText);

  //     const submitEventMock = {
  //       preventDefault: () => null,
  //     };
  //     const props = {
  //       onSave: () => null,
  //     };

  //     wrapped.setProps(props);
  //     wrapped.find('form').simulate('submit', submitEventMock);
  //     wrapped.update();
  //     expect(wrapped.find('input[type="email"]').prop('value')).toBe('');
  //   });

  it('renders a Button', () => {
    expect(newLogInForm.find(Button)).toHaveLength(1);
  });

  it('renders a LinkGoogle', () => {
    expect(newLogInForm.find(LinkGoogle)).toHaveLength(1);
  });

  it('renders five NavLink', () => {
    expect(newLogInForm.find(NavLink)).toHaveLength(1);
  });

  it('renders two div', () => {
    expect(newLogInForm.find('div')).toHaveLength(2);
  });

  it('renders one form', () => {
    expect(newLogInForm.find('form')).toHaveLength(1);
  });

  it('renders two label', () => {
    expect(newLogInForm.find('label')).toHaveLength(2);
  });

  it('renders one input', () => {
    expect(newLogInForm.find('input[type="email"]')).toHaveLength(1);
  });

  it('renders one input', () => {
    expect(newLogInForm.find('input[type="password"]')).toHaveLength(1);
  });

  it('renders one p', () => {
    expect(newLogInForm.find('p')).toHaveLength(1);
  });

  // -- Does not work!!! --
  // it('renders two span', () => {
  //   expect(newLogInForm.find('span')).toHaveLength(2);
  // });

  // describe('callback', () => {
  // let wrapped;

  // beforeEach(() => {
  //   wrapped = shallow(<LogInForm />);
  // });

  // it('onSave prop has been called on submit', () => {
  //   const submitEventMock = {
  //     preventDefault: () => null,
  //   };

  //   const onSavePropMock = jest.fn();
  //   const props = {
  //     onSave: onSavePropMock,
  //   };

  //   wrapped.setProps(props);
  //   wrapped.find('form').simulate('submit', submitEventMock);

  //   expect(onSavePropMock.mock.calls.length).toBe(1);
  // });
  // });

  describe('testing login', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<LogInForm />);
    });

    it('email check', () => {
      wrapper.find('input[type="email"]').simulate('change', {
        target: { name: 'email', value: 'test@.ukr.net' },
      });

      expect(wrapper.state('email')).toEqual('test@.ukr.net');
    });

    it('password check', () => {
      wrapper.find('input[type="password"]').simulate('change', {
        target: { name: 'password', value: '123456' },
      });

      expect(wrapper.state('password')).toEqual('123456');
    });
  });
});
