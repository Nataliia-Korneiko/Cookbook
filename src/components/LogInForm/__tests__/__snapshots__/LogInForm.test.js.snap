// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogInForm component renders a properly 1`] = `ShallowWrapper {}`;
